### ISSUE TITLE
mix works inconsistently

### ISSUE THREAD

**Issue by iamstarkov** [bem/bh#86]
`mix` mix classes and not `tags`, `attrs` and other `mixes` and it's a consistency problem.
## tags

I want to use `link` block as mixin and not to hack it every time, so I want to write this way, but it's not working:

``` js
{
    block: 'media',
    elem: 'link',
    mix: [{ block: 'link', url: 'http://github.com/' }],
    content: [
        {
            elem: 'icon',
            url: 'duck.png'
        },
        {
            elem: 'title',
            content: 'comment'
        }
    ]
}
```

So I have to write this ugly code:

``` js
{
    block: 'link',
    mix: [{ block: 'media', elem: 'link' }],
    url: 'http://github.com/',
    content: [
        {
            block: 'media', elem: 'icon',
            url: 'duck.png'
        },
        {
            block: 'media', elem: 'title',
            content: 'comment'
        },
    ]
}
```
## attrs

Then, I want to use one block for analytics tracking, and it have `onclick` attrs and goal option. I want to mix it and expect to see `onclick` attr on parent block like this way:

**i-track.bh.js:**

``` js
bh.match('i-track', function (ctx, json) {
    ctx.attr('onclick', 'track("' + json.goal + '")');
});
```

and use it like this:

``` js
{   
    block: 'link',
    mix: [{ block: 'i-track', goal: 'clck_goal' }],
    url: 'http://github.com/',
    content: 'Tracking link'
}
```

But it's not working too.
## other mixes

I want to use custom font on my site. So I have one block `i-face` with mods for each custom font, but in reality only one font is being used in project, so it's reasonable to create shortcut block `textbook`, which have it's own mix with concrete `textbook` font.

In other words, I don't want to write `mix: [{ block: 'i-font', mods: { face: 'textbook' }}]` every time, because it's hard to remember and want to short it to this: `mix: [{ block: 'textbook' }]`, where textbook block have such `bh` file:

``` js
bh.match('textbook', function (ctx, json) {
    ctx.mix([{ block: 'i-font', mods: { face: 'textbook' }}]);
});
```

And finally it's not working at all.

**Comment by iamstarkov** [comment]
not relevant anymore
